### PR TITLE
Add Change Notification Preferences Dialog

### PR DESCRIPTION
**Pull request by evancrow**
Currently only attached to the Cookie Cutter onboarding flow.

### How was this tested?
- [x] I tested this manually

### Screenshots and screen recordings

https://user-images.githubusercontent.com/37356984/166591295-1ef2a97f-60b9-4f28-8c01-15c4692204fc.mp4

## Issues addressed
Fixed: #3500

cc @hwiatneeva

### CODE REVIEW

**Comment by hwiatneeva** [comment]
a neat ios feature - good to know about it. Although we're still figuring out whether or not we'll have "Remind me later" as a feature, yet, the implementation as-is LGTM. Thanks @evancrow !

**Comment by evancrow** [comment]
> a neat ios feature - good to know about it. Although we're still figuring out whether or not we'll have "Remind me later" as a feature, yet, the implementation as-is LGTM. Thanks @evancrow !

Yeah just using this as a placeholder for now until we figure out what the plan is.

File: Client/Frontend/Notifications/NotificationPermissionHelper.swift [hunk 1]
```diff
@@ -76,10 +77,14 @@ class NotificationPermissionHelper {
...
 
                     self.requestPermissionFromSystem(completion: completion, callSite: callSite)
-                } else if openSettingsIfNeeded {
+                } else if showChangeInSettingsDialogIfNeeded,
+                    let bvc = bvc ?? SceneDelegate.getBVCOrNil()
+                {
                     /// If we can't show the iOS system notification because the user denied our first request,
-                    /// this will take them to system settings to enable notifications there.
-                    SystemsHelper.openSystemSettingsNeevaPage()
+                    /// ask the suer if they would like to change that in settings.
```
Comment (chung-lai) [inline Client/Frontend/Notifications/NotificationPermissionHelper.swift:84]: nit: `suer` -> `user`

**Review by chung-lai — APPROVED** [review]
LGTM, thanks!

**Review by darinwf — APPROVED** [review]
LGTM!

**Pull request merged**